pddl/logic/base.py

```diff
@@ -115,31 +115,34 @@ class Atomic(Formula):
     """Atomic formula."""
 
 
-class MonotoneOp(type):
+class BinaryOpMetaclass(type):
     """Metaclass to simplify monotone operator instantiations."""
 
     _absorbing: Optional[Formula] = None
+    idempotency: bool = False
 
     def __call__(cls, *args, **kwargs):
         """Init the subclass object."""
         operands = _simplify_monotone_op_operands(cls, *args)
-        if len(operands) == 1:
+        if len(operands) == 1 and cls.idempotency:
             return operands[0]
```

Comment:
great, but I expected also line 126, the call to  `_simplify_monotone_op_operands`, to change.

Now this gives me:
```python
from pddl.logic.functions import Minus, NumericValue
a1 = NumericValue(1)
m = Minus(a1, a1, a1)
m.operands  # returns a tuple of one element: `(a1, )`
```

Reply:
right! This has been fixed in [2682e2f](https://github.com/AI-Planning/pddl/pull/92/commits/2682e2f51f89b920f697896ae485e58b6d4f3296)